### ISSUE TITLE
Horizon Zero Dawn: Low resolution textures fixed with VKD3D_CONFIG=no_upload_hvv

### ISSUE THREAD

**Issue by kscherer** [HansKristian-Work/vkd3d-proton#992]
The game runs well but looks terrible due to low resolution textures.

Ubuntu 21.04 kernel 5.13.0 RADV 21.2.2 (same issue with RADV 22)
GPU: RX 580 4GB - Medium graphics With SMAA (no motion blur). Same issue with upscaling.
Proton-GE 7.1-GE-2 (Same issue with Proton Experimental and previous Proton GE versions 7.0rc and 6.21)

I actually get better performance (35 fps vs 45 fps) in the benchmark with VKD3D_CONFIG=no_upload_hvv

Got the hint to try this option from Issue #864 because Death Stranding uses the same engine as HZD

![HZD_no_upload_hvv_2](https://user-images.githubusercontent.com/347043/153005854-c44e7966-d3c8-4d3c-958a-428582fbc1cb.png)
![HZD_map_texture](https://user-images.githubusercontent.com/347043/153005867-95375b6b-bb21-4772-a169-6b1b5794f665.png)

**Comment by doitsujin** [comment]
Maybe we should restrict HVV usage to GPUs with ≥8GB of vram and limit the amount to allocate from it further?

Although this *should* be transparent to the application since our own allocations shouldn't affect the budget reported by the same process.

**Comment by HansKristian-Work** [comment]
It will be allocated in DEVICE_LOCAL, so it should affect the budget. I guess we can try just having non-HVV default for 4GB in general.

**Comment by kscherer** [comment]
Yes it seems obvious in hindsight but it took me a long time to find the right option. I wish I could buy a GPU with more VRAM at a reasonable price. Even the new GPUs like the RX6500 have 4GB VRAM and almost same performance as RX580. Hopefully this helps others out there with low VRAM GPUs.

**Comment by doitsujin** [comment]
> It will be allocated in DEVICE_LOCAL, so it should affect the budget. I guess we can try just having non-HVV default for 4GB in general, there doesn't seem to be any tangible benefit anyways.

Isn't the whole point of the budget that allocations made by the same process **do not** affect the reported budget?
